public/pages/Findings/containers/Findings/Findings.tsx

```diff
@@ -157,44 +157,57 @@ class Findings extends Component<FindingsProps, FindingsState> {
     this.setState({ loading: true });
     const { findingsService, detectorService, notifications } = this.props;
     try {
-      const detectorsRes = await detectorService.getDetectors();
-      if (detectorsRes.ok) {
-        const detectors = detectorsRes.response.hits.hits;
+      // Get all the Findings logic
+      const findingRes = await findingsService.getFindings();
+
+      if (findingRes.ok) {
         const ruleIds = new Set<string>();
         let findings: FindingItemType[] = [];
 
-        const detectorId = this.props.match.params['detectorId'];
-        for (let detector of detectors) {
-          if (!detectorId || detector._id === detectorId) {
-            const findingRes = await findingsService.getFindings({ detectorId: detector._id });
-
-            if (findingRes.ok) {
-              const detectorFindings: FindingItemType[] = findingRes.response.findings.map(
-                (finding) => {
-                  finding.queries.forEach((rule) => ruleIds.add(rule.id));
-                  return {
-                    ...finding,
-                    detectorName: detector._source.name,
-                    logType: detector._source.detector_type,
-                    detector: detector,
-                  };
-                }
-              );
-              findings = findings.concat(detectorFindings);
-            } else {
-              errorNotificationToast(notifications, 'retrieve', 'findings', findingRes.error);
+        // Assuming getDetectors returns an array of detectors
+        const detectorsRes = await detectorService.getDetectors();
+        if (detectorsRes.ok) {
+          const detectors = detectorsRes.response.hits.hits;
+          const detectorInfoMap = new Map();
+
+          detectors.forEach((detector) => {
+            detectorInfoMap.set(detector._id, {
+              logType: detector._source.detector_type,
+              name: detector._source.name,
+              detector: detector,
+            });
+          });
+          const detectorFindingsPromises = findingRes.response.findings.map(async (finding) => {
+            try {
+              // Use detectorInfoMap to get detector information
+              const detectorInfo = detectorInfoMap.get(finding.detectorId);
+              finding.queries.forEach((rule) => ruleIds.add(rule.id));
+              return {
+                ...finding,
+                detectorName: detectorInfo ? detectorInfo.name : 'Unknown',
+                logType: detectorInfo ? detectorInfo.logType : 'Unknown',
```

Comment:
Nitpick: Would it make sense to use the `DEFAULT_EMPTY_DATA` constant ([link](https://github.com/opensearch-project/security-analytics-dashboards-plugin/blob/7837a99e26f356542a1006a4329eecbcb3fe44ce/public/utils/constants.ts#L19)) here instead of `Unknown`?

Reply:
Also, do you happen to know if there are instances when we'd expect the `logType` or `detectorName` to be undefined for a finding? If not, I'm wondering if it would be worthwhile to add a log of some kind here as that seems like it might be an odd anomaly.